src/hooks/bookmarks/use-is-bookmarked.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { ApiTutorial } from 'lib/learn-client/api/api-types'
+import { useAllBookmarks } from './use-all-bookmarks'
+
+interface UseIsBookmarkedOptions {
+	tutorialId: ApiTutorial['id']
+}
+
+interface UseIsBookmarkedResult extends Omit<UseQueryResult, 'data'> {
+	isBookmarked: null | boolean
+}
+
+/**
+ * Handles checking if there is a bookmark for the given `tutorialId`.
+ */
+const useIsBookmarked = ({
+	tutorialId,
+}: UseIsBookmarkedOptions): UseIsBookmarkedResult => {
+	// Load up all bookmarks, since we know it primes queries & handles auth
+	const { bookmarks, isFetched: bookmarksAreFetched } = useAllBookmarks()
```

Comment:
Hmmm I would expect us to just fetch once single tutorial here. Trying to think through this.

Reply:
Based on how this hook is intended to be used (once for each `TutorialCard` or `TutorialMeta`), that would require doing a fetch for each card/meta. So if there are 20 cards on a page, that means 20 fetches. This approach reduces it to one fetch in total.

Reply:
Right, for that use case I figured that the bookmark component itself wouldn't call the `isBookmarked` hook, and that data would be passed down from higher. 

But just based on the naming of this hook, I would expect it to query just a single bookmark.

Reply:
While we could pass data down, one of the primary purposes of using the hooks is so we can avoid props drilling. The hook does return the value one single bookmark, it just makes sure the data it needs to do so is already present. And reduces the number of fetches made as a side effect.

Reply:
Well the tutorial meta only cares about one tutorial, not multiple. So the cards are where the multiple fetches come into play. Is there a way to prime a global cache in a way, such that if we make a call for a particular tutorial on say a product landing page, that data would be cached for the tutorial and collection page views? 

Or is this approach what that is intended to do in a way?

Reply:
The query priming approach wasn't working right away and needs some more time to explore, so that's another reason why optimizing this hook has been deferred for now. Once the individual query `bookmark` priming is worked out, then we can step `useIsBookmarked` toward no longer fetching all bookmarks to "prime" itself.

Reply:
In dev mode, I'm seeing that every time a bookmark is created or deleted the 'bookmarks' endpoint is being called with this set-up. Having all bookmarks called when just one needs to be updated doesn't seem like the behavior we want.

What was happening with the individual priming? Should we wait to figure that out before finishing up these hooks?

Reply:
Sorry if I wasn't clear! I want to overall optimize and also resolve the query priming in a separate PR since everything is working as needed here.